### PR TITLE
[T4A3][T14-B1] Tan Min Rong Samson

### PR DESCRIPTION
**Pull request by samsontmr**
Ready for Review.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,39 +6,22 @@
...
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
+        super(address, isPrivate, ADDRESS_VALIDATION_REGEX, MESSAGE_ADDRESS_CONSTRAINTS);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java:21]: :+1:

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/**
+ * 
+ */
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Contains methods and variables common to Phone, Email and Address classes
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:9]: :+1:  it is not necessary to include details about the child classes here.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+public class Contact {
+    public final String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:13]: try to make it private. in general, start off with private members, then use protected or public only if absolutely necessary.

**Comment by okkhoy** [comment]
@samsontmr good job on this PR :+1: 
Some comments added. Please ack & close the PR after reading comments.

**Comment by samsontmr** [comment]
@okkhoy Acknowledged, thanks!

**Pull request closed without merging**